include/amqqueue.hrl

```diff
@@ -51,16 +51,16 @@
          (?is_amqqueue_v1(Q) andalso
           ?amqqueue_v1_field_state(Q) =:= State))).
 
--define(amqqueue_v1_type, classic).
+-define(amqqueue_v1_type, rabbit_classic_queue).
```

Comment:
In 3.7.x, we can change the value of `?amqqueue_v1_type` to `rabbit_classic_queue` and be done with it.

In 3.8.x, in addition to the above, we can update the `amqqueue` and `amqqueue_v1` modules to accept the value `classic` in the `new*()` functions and, where relevant, convert it to `?amqqueue_v1_type`. The record will never have `classic` stored inside.

When people switch to using the `amqqueue` modules, they can't change the value of the `type` field because records are private/opaque in 3.8.x.

What do you think?

Reply:
ok I've updated the PR with what I think you are describing :) can you check the last commit?

Reply:
I added another review which points to an unmodified line and GitHub doesn't let me attach the comment to that line.